packages/mint-components/src/components/sqm-portal-email-verification/usePortalEmailVerification.tsx

```diff
@@ -34,6 +54,66 @@ export function usePortalEmailVerification(props: PortalEmailVerification) {
       setSuccess(true);
   };
 
+  async function checkVerification() {
+    const data =
+      (await getVerificationStatus()) as ManagedIdentitySessionResult;
+    if (data?.managedIdentitySession?.emailVerified) {
+      clearInterval(timerRef.current);
+      return navigation.push({
+        pathname: props.redirectPath,
+        search: urlParams.toString() && "?" + urlParams.toString(),
+      });
+    }
+    resyncTimers(10);
+  }
+
+  const startTimer = (countdown: number = 10000) =>
+    setInterval(checkVerification, countdown);
+  const startCountdown = (countdown: number) =>
+    setInterval(() => setCountdown(countdown), 1000);
+
+  function resyncTimers(restartCountdown?: number) {
+    // Clear existing timers
+    clearInterval(countdownRef.current);
+    clearInterval(timerRef.current);
+
+    // Do nothing if the tab is currently hidden
+    if (document.visibilityState === "hidden") return;
+
+    // Restart the timer if verification has just been checked
+    if (restartCountdown) setCountdown(restartCountdown);
+
+    // Start timers from either the beginning or based on the previous value
+    countdownRef.current = startCountdown(
+      restartCountdown - 1 || countdown - 1
+    );
+    timerRef.current = startTimer(restartCountdown * 1000 || countdown * 1000);
+  }
+
+  const resyncTimersCallback = useCallback(() => resyncTimers(), [countdown]);
+
+  // Refetch validation status timer
+  useEffect(() => {
+    if (!timerRef.current) {
+      checkVerification();
+      timerRef.current = startTimer();
+    }
+    return () => {
+      clearInterval(timerRef.current);
+    };
+  }, []);
+
+  // Countdown timer
+  useEffect(() => {
+    if (countdown > 0) countdownRef.current = startCountdown(countdown - 1);
```

Comment:
Is this safe? There's a new interval every time the countdown ticks but it'll only clean up the last one

Reply:
Not 100% sure, but it now clears the interval each count just to be safe.